contracts/social-tips/src/contract.rs

```diff
@@ -0,0 +1,1346 @@
+                to_address: replaced_tip.sender.to_string(),
+            });
+        }
+
+        Ok(response)
+    };
+}
+
+pub fn claim_tips(
+    deps: DepsMut<DesmosQuery>,
+    info: MessageInfo,
+) -> Result<Response<DesmosMsg>, ContractError> {
+    let mut coins = Vec::<Coin>::new();
+    let querier = ProfilesQuerier::new(deps.querier.deref());
+    let pending_tips_map = pending_tips();
+
+    for app_link_result in
+        querier.iterate_application_links(Some(info.sender.clone()), None, None, 10)
+    {
+        let app_link = app_link_result?;
+        if app_link.state == ApplicationLinkState::VerificationSuccess {
+            let mut pending_tips = pending_tips_map
+                .prefix((
+                    app_link.data.application.clone(),
+                    app_link.data.username.clone(),
+                ))
+                .range(deps.storage, None, None, Order::Ascending)
+                .collect::<StdResult<Vec<_>>>()?;
+
+            for (sender, mut pending_tip) in pending_tips.drain(0..) {
+                pending_tip
+                    .amount
+                    .drain(0..)
+                    .for_each(|coin| coins.push(coin));
+                pending_tips_map.remove(
+                    deps.storage,
+                    (
+                        app_link.data.application.clone(),
+                        app_link.data.username.clone(),
+                        sender,
+                    ),
+                )?;
+            }
+        }
+    }
+
+    let merged_coins = sum_coins_sorted(coins)?;
+
+    if merged_coins.is_empty() {
+        return Err(ContractError::NoTipsAvailable {
+            user: info.sender.to_string(),
+        });
+    }
+
+    Ok(Response::new()
+        .add_attribute(ATTRIBUTE_ACTION, ACTION_CLAIM_PENDING_TIPS)
+        .add_attribute(ATTRIBUTE_TIP_RECEIVER, &info.sender)
+        .add_attribute(ATTRIBUTE_TIP_AMOUNT, serialize_coins(&merged_coins))
+        .add_message(BankMsg::Send {
+            amount: merged_coins,
+            to_address: info.sender.to_string(),
+        }))
+}
+
+fn update_admin(
+    deps: DepsMut<DesmosQuery>,
+    new_admin: String,
+) -> Result<Response<DesmosMsg>, ContractError> {
+    let new_admin_addr = deps.api.addr_validate(&new_admin)?;
+
+    CONFIG.update(deps.storage, |mut config| {
+        if config.admin != new_admin_addr {
+            return Err(ContractError::Unauthorized {});
+        }
+
+        config.admin = new_admin_addr;
+        Ok(config)
+    })?;
+
+    Ok(Response::new()
+        .add_attribute(ATTRIBUTE_ACTION, ACTION_UPDATE_ADMIN)
+        .add_attribute(ATTRIBUTE_NEW_ADMIN, &new_admin))
+}
+
+fn update_max_pending_tips(
+    deps: DepsMut<DesmosQuery>,
+    info: MessageInfo,
+    value: u16,
+) -> Result<Response<DesmosMsg>, ContractError> {
+    let mut config = CONFIG.load(deps.storage)?;
+
+    if config.admin != info.sender {
+        return Err(ContractError::Unauthorized {});
+    }
```

Comment:
We can have a function to check if the sender is admin or not. Then it can be replaced to something like:
```
check_admin(&deps.storage, sender)?;
```